tests/test_json_schema_to_grammar.py

```diff
@@ -0,0 +1,325 @@
+import json
+
+import pytest
+from jsonschema import validate
+
+from guidance._grammar import GrammarFunction
+from guidance._json_schema_to_grammar import json_schema_to_grammar
+from guidance._parser import EarleyCommitParser, ParserException
+
+
+def to_compact_json(target: any) -> str:
+    # See 'Compact Encoding':
+    # https://docs.python.org/3/library/json.html
+    # Since this is ultimately about the generated
+    # output, we don't need to worry about pretty printing
+    # and whitespace
+    return json.dumps(target, separators=(",", ":"))
```

Comment:
Seems reasonable to generate compact JSONs, which users can then pretty-print themselves.